crates/nix_rs/src/flake/schema.rs

```diff
@@ -16,15 +16,22 @@ use super::{
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct FlakeSchema {
     pub system: System,
-    pub packages: BTreeMap<String, Val>,
-    pub legacy_packages: BTreeMap<String, Val>,
-    pub devshells: BTreeMap<String, Val>,
-    pub checks: BTreeMap<String, Val>,
-    pub apps: BTreeMap<String, Val>,
-    pub formatter: Option<Val>,
+    pub packages: BTreeMap<String, Leaf>,
+    pub legacy_packages: BTreeMap<String, Leaf>,
+    pub devshells: BTreeMap<String, Leaf>,
+    pub checks: BTreeMap<String, Leaf>,
+    pub apps: BTreeMap<String, Leaf>,
+    pub formatter: Option<Leaf>,
+    pub nixos_configurations: BTreeMap<String, Leaf>,
+    pub darwin_configurations: BTreeMap<String, Leaf>,
```

Comment:
At this point, I'm wondering - can't these fields be collapsed into one, of type `BTreeMap<String, BTreeMap<String, Leaf>>` or some such thing to operate on the schemas *generically*?

It may also enable rendering the per-schema stuff generically in `crates/omnix-cli/src/command/show.rs`.

Reply:
But let's get this PR working first. I can take a look at making it generic before merging.